src/runtimes/node/parser/exports.ts

```diff
@@ -0,0 +1,85 @@
+import { CallExpression, Statement } from '@babel/types'
+
+import type { ISCExport } from '../in_source_config'
+
+import { isModuleExports } from './helpers'
+
+// Finds the main handler export in an AST.
+const getMainExport = (nodes: Statement[]) => {
+  let handlerExport: ISCExport[] = []
+
+  nodes.some((node) => {
+    const esmExports = getMainExportFromESM(node)
+
+    if (esmExports.length !== 0) {
+      handlerExport = esmExports
+
+      return true
+    }
+
+    const cjsExports = getMainExportFromCJS(node)
+
+    if (cjsExports.length !== 0) {
+      handlerExport = cjsExports
+
+      return true
+    }
+
+    return false
+  })
+
+  return handlerExport
+}
+
+// Finds the main handler export in a CJS AST.
+const getMainExportFromCJS = (node: Statement) => {
+  const handlerPath = ['module', 'exports', 'handler']
+
+  if (!isModuleExports(node, handlerPath) || node.expression.right.type !== 'CallExpression') {
+    return []
+  }
+
+  return getExportsFromCallExpression(node.expression.right)
+}
+
+// Finds the main handler export in an ESM AST.
+// eslint-disable-next-line complexity
+const getMainExportFromESM = (node: Statement) => {
+  if (node.type !== 'ExportNamedDeclaration' || node.exportKind !== 'value') {
+    return []
+  }
+
+  const { declaration } = node
+
+  if (!declaration || declaration.type !== 'VariableDeclaration') {
+    return []
+  }
+
+  const handlerDeclaration = declaration.declarations.find((childDeclaration) => {
+    const { id, type } = childDeclaration
+
+    return type === 'VariableDeclarator' && id.type === 'Identifier' && id.name === 'handler'
+  })
+
+  if (handlerDeclaration?.init?.type !== 'CallExpression') {
+    return []
+  }
+
+  const exports = getExportsFromCallExpression(handlerDeclaration.init)
```

Comment:
there can only ever be one export called `handler`. are we thinking in multiples here because to future-proof this for potential additional exports?

Reply:
Yes, exactly. The overhead in complexity feels really small, and we prepare ourselves to accommodate different patterns.